### PR TITLE
Update Land DA container to Spack-Stack 1.6.0

### LINKED ISSUE #143: Need to update container capability
- Currently, the container option support the previous version (v1.2.0).
- Since the structure and capabilities of the land-DA workflow were significantly changed, the container part should be updated to support the next release (v2.0.0).

### PR DESCRIPTION
**Pull request by EdwardSnyder-NOAA**
## Description

This PR updates or adds files to support updating the Land DA container to spack-stack 1.6.0. Major changes include adding modulefile task files for singularity, added a singularity land_analysis file, and document changes.

The Land DA container was built with this PR using this [landda dockerfile](https://github.com/NOAA-EPIC/ufs-containers/blob/feature/unified-1.6.0/Docker/landda/Dockerfile.ubuntu22.04-intel21.10-ue160-landda) which inherits this [spack-stack dockerfile](https://github.com/NOAA-EPIC/ufs-containers/blob/feature/unified-1.6.0/Docker/unified/Dockerfile.ubuntu22.04-intel-unified). Testing was done with this container using the land_analysis test described in the docs which passed on GCP, AWS, and Hercules.

UPDATE:
The latest Land DA container was built with the updated wm hash, which required updates to the [landda dockerfile](https://github.com/NOAA-EPIC/ufs-containers/blob/feature/unified-1.6.0/Docker/landda/Dockerfile.ubuntu22.04-intel21.10-ue160-fms202401-landda) and [spack-stack dockerfile](https://github.com/NOAA-EPIC/ufs-containers/blob/feature/unified-1.6.0/Docker/unified/Dockerfile.ubuntu22.04-intel-unified-fms2024.01). Here is the [setup_container.sh](https://github.com/NOAA-EPIC/ufs-containers/blob/feature/unified-1.6.0/Docker/landda/setup_container.sh) script that is used in this process. Lastly, the ctest Dockerfile was tested with our Jenkins pipeline and passes without any problems.

### Anticipated changes to regression tests:
- [ ] Is baseline change expected ? 

## Subcomponents involved:
- [ ] apply_incr.fd (NOAA-PSL/land-apply_jedi_incr)
- [ ] ufs_model.fd (ufs-community/ufs-weather-model)
- [ ] none

### Testing (for CM's):
- RDHPCS
    - [ ] Hera
    - [ ] Orion
    - [ ] Hercules
- CI
  - [ ] Completed
- WE2E
  - [ ] Completed
- PW-Clouds
  - [ ] AWS
  - [ ] AZURE
  - [ ] GCP

### CODE REVIEW

**Comment by jkbk2004** [comment]
@EdwardSnyder-NOAA do you think we need to update https://github.com/EdwardSnyder-NOAA/land-DA_workflow/blob/ss160-container-update/sorc/test/ci/Dockerfile ?

**Comment by EdwardSnyder-NOAA** [comment]
Yes, @jkbk2004 I'll need to update that file. I got it to work but had to make a number of changes that I think could be resolved in a more elegant way. I'll try to get those changes in today. 

Just a reminder this PR is for the previous ufs model hash. I'm still working on building the spack-stack with the proper modules to support the new ufs model hash.

**Comment by EdwardSnyder-NOAA** [comment]
The container has been updated to work with the new ufs wm hash. All the scripts are in order and it runs successfully on AWS (`/contrib/Edward.Snyder/landda-container/aug-update/wm-hash-update/updated/land-DA_workflow`):
```
[Edward.Snyder@aws-202 parm]$ rocotostat -w land_analysis.xml -d land_analysis.db
       CYCLE                    TASK                       JOBID               STATE         EXIT STATUS     TRIES      DURATION
================================================================================================================================
200001030000                prep_obs                          54           SUCCEEDED                   0         1           9.0
200001030000                pre_anal                          55           SUCCEEDED                   0         1          17.0
200001030000                analysis                          57           SUCCEEDED                   0         1          26.0
200001030000               post_anal                          58           SUCCEEDED                   0         1           9.0
200001030000                forecast                          59           SUCCEEDED                   0         1          40.0
200001030000              plot_stats                          60           SUCCEEDED                   0         1          59.0
================================================================================================================================
200001040000                prep_obs                          56           SUCCEEDED                   0         1           9.0
200001040000                pre_anal                          61           SUCCEEDED                   0         1           8.0
200001040000                analysis                          62           SUCCEEDED                   0         1          19.0
200001040000               post_anal                          63           SUCCEEDED                   0         1           8.0
200001040000                forecast                          64           SUCCEEDED                   0         1          38.0
200001040000              plot_stats                          65           SUCCEEDED                   0         1          25.0
```
The ctest Dockerfile also works, which was tested with our Jenkins pipeline:
```
#13 57.04 100% tests passed, 0 tests failed out of 4
#13 57.04 
#13 57.04 Total Test time (real) =  55.52 sec
#13 DONE 70.2s`

File: ush/plot_forecast_restart.py [hunk 1]
```diff
@@ -50,6 +50,8 @@ def main():
         cartopy.config['data_dir']='/scratch2/NAGAPE/epic/UFS_Land-DA_Dev/inputs/NaturalEarth'
     elif yaml_data['machine']=='orion' or yaml_data['machine']=='hercules':
         cartopy.config['data_dir']='/work/noaa/epic/UFS_Land-DA_Dev/inputs/NaturalEarth'
+    elif yaml_data['machine']=='singularity':
+                cartopy.config['data_dir']='SINGULARITY_WORKING_DIR/land-DA_workflow/fix/NaturalEarth'
```
Comment (chan-hoo) [inline ush/plot_forecast_restart.py]: @EdwardSnyder-NOAA, is this indentation correct? It looks indented twice.

Comment (EdwardSnyder-NOAA) [reply ush/plot_forecast_restart.py]: this has been fixed. Thanks!

**Review by chan-hoo — APPROVED** [review]

**Pull request merged**